src/module_PI/Raspberry_PI/main/PacketMessageTranslator.java

```diff
@@ -0,0 +1,110 @@
+		messageTable.put(150, "Button object sucessfully created");
+		messageTable.put(151, "Button has been pressed");
+		messageTable.put(200, "Buzzer object has been created");
+		messageTable.put(201, "Buzzer has started to sound");
+		messageTable.put(202, "Buzzer has stopped sounding");
+		messageTable.put(250, "Component object sucessfully created");
+		messageTable.put(300, "ComponentTester default constructor called");
+		messageTable.put(301, "ComponentTester object sucessfully created");
+		messageTable.put(350, "Laser object sucessfully created");
+		messageTable.put(351, "Laser has been turned on");
+		messageTable.put(352, "Laser has been turned off");
+		messageTable.put(400, "LED object sucessfully created");
+		messageTable.put(401, "LED has been turned on");
+		messageTable.put(402, "LED has been turned off");
+		messageTable.put(403, "LED flashing");
+		messageTable.put(404, "LED toggling");
+		messageTable.put(450, "Photoresistor object has been successfully created");
+		messageTable.put(451, "Photoresistor taking reading");
+	}
+	
+	public String translate(int packetDetails){
+		if(isIntInScope(packetDetails)) throw new IllegalArgumentException("invalid packet int: " + packetDetails);
+		int correctedInt = parsePacketDetailsFromFullPacketInt(packetDetails);
+		if(!messageTable.containsKey(correctedInt)) throw new IllegalArgumentException("packetDetails " + packetDetails + " not in messageTable");
+		String packetMessage = messageTable.get(correctedInt);
+		return packetMessage;
+	}
+	
+	private int parsePacketDetailsFromFullPacketInt(int packetAsInt){
+		int packetModThisIsCorrectLength = (int) (Math.pow(10, NUM_OF_BYTES_IN_PACKET_DETAILS));
+		return packetAsInt % packetModThisIsCorrectLength;
+	}
+	
+	private boolean isIntInScope(int packetDetails){
+		return (packetDetails < 1 
+				|| packetDetails > MAX_PACKET_DETAILS_INT
+				|| packetDetails > MAX_DETAILS_ONLY_INT && packetDetails < MIN_FULL_PACKET_INT);
+	}
+	
+	public String translate(String packetDetailsAsString){
+		if(isStringNumeric(packetDetailsAsString)) return translate(Integer.parseInt(packetDetailsAsString));
+		throw new IllegalArgumentException(packetDetailsAsString + " is not an integer");
+	}
+	
+	private boolean isStringNumeric(String numericString){
+		return numericString.matches("^\\d+$");
+	}
+	
+	public String translate(byte[] packetAsByteArray){
+		if(packetIsInvalidLength(packetAsByteArray)) throw new IllegalArgumentException(packetAsByteArray.length + " is an invalid byte length");
+		return packetByteArrayToMessage(packetAsByteArray);
+	}
+	
+	private boolean packetIsInvalidLength(byte[] packetAsByteArray){
+		return(packetAsByteArray.length != NUM_OF_BYTES_IN_PACKET 
+				&& packetAsByteArray.length != NUM_OF_BYTES_IN_PACKET_DETAILS);
+	}
+	
+	private String packetByteArrayToMessage(byte[] packetAsByteArray){
+		int parsedDetailsIntFromByteArray = parseDetailsIntFromByteArray(packetAsByteArray);
+		return translate(parsedDetailsIntFromByteArray);
```

Comment:
Another instance where you could combine these into a single line if you wanted:
```java
return translate(parseDetailsIntFromByteArray(packetAsByteArray));
```